route.go

```diff
@@ -64,7 +64,6 @@ func (r *Route) Match(req *http.Request, match *RouteMatch) bool {
 				match.MatchErr = nil
 			}
 
-			matchErr = nil
```

Comment:
This change feels a bit outside the scope of this particular PR.

Reply:
Golint pointed out that this is a `deadcode`, that's why I've removed. I can revert this part and mark it with `// nolint` for now.